langkit/documentation.py

```diff
@@ -248,6 +248,14 @@ def instantiate_templates(doc_dict: Dict[str, str]) -> DocDatabase:
         Raised when introspection functions (``${ctx.lib_name}.Introspection``)
         are improperly used to evaluate a node data (field or property).
     """,
+    'langkit.introspection.bad_type_error': """
+        Raised when introspection functions (``${ctx.lib_name}.Introspection``)
+        are given values with inconsistent types.
+    """,
+    'langkit.introspection.out_of_bounds_error': """
+        Raised when introspection functions (``${ctx.lib_name}.Introspection``)
+        are queried an out-of-bound index access on an array.
```

Comment:
1. I'm not sure of the value of having exceptions with so specific purposes, what use case do you see against just one for introspection errors?
2. The formulation of this message is weird: "are passed an index that generates an out of bound access to an array".

EDIT: About 1., see comment above about type errors

Reply:
Thinking more about it, the description is too specific. I’ll try “are passed an out of bounds index”. As discussed live, we’ll keep this exception, but merge most others into the “bad type” one.